scripts/zkai_attention_kv_stwo_binary_typed_proof_accounting_gate.py

```
@@ -0,0 +1,685 @@
+        raise BinaryTypedProofAccountingGateError(f"{role} local accounting field drift")
+    if accounting["format_domain"] != ACCOUNTING_DOMAIN:
+        raise BinaryTypedProofAccountingGateError(f"{role} accounting domain drift")
+    if accounting["format_version"] != ACCOUNTING_FORMAT_VERSION:
+        raise BinaryTypedProofAccountingGateError(f"{role} accounting format version drift")
+    if accounting["upstream_stwo_serialization_status"] != UPSTREAM_SERIALIZATION_STATUS:
+        raise BinaryTypedProofAccountingGateError(f"{role} upstream serialization status drift")
+    records = accounting["records"]
+    if not isinstance(records, list) or len(records) != len(EXPECTED_RECORD_PATHS):
+        raise BinaryTypedProofAccountingGateError(f"{role} accounting record count drift")
+    if accounting["record_count"] != len(records):
+        raise BinaryTypedProofAccountingGateError(f"{role} record_count drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)